lib/plugins/nwc/src/model.rs

```diff
@@ -0,0 +1,172 @@
+use nostr_sdk::{nips::nip47::NostrWalletConnectURI, PublicKey};
+use serde::{Deserialize, Serialize};
+
+use crate::{utils, DEFAULT_RELAY_URLS};
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct NwcConfig {
+    /// A list of default relay urls to add per connection
+    pub relay_urls: Option<Vec<String>>,
+    /// Custom Nostr secret key (hex-encoded) for the wallet node
+    pub secret_key_hex: Option<String>,
+}
+
+impl NwcConfig {
+    pub fn relays(&self) -> Vec<String> {
+        self.relay_urls
+            .clone()
+            .unwrap_or(DEFAULT_RELAY_URLS.iter().map(|s| s.to_string()).collect())
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub(crate) struct PeriodicBudgetInner {
+    pub used_budget_sat: u64,
+    pub max_budget_sat: u64,
+    /// The duration of the budget's period
+    /// ## Dev Note:
+    /// If the renewal time is less than [crate::MIN_REFRESH_INTERVAL_SEC] seconds,
+    /// then it will take at most [crate::MIN_REFRESH_INTERVAL_SEC] seconds in order for the
+    /// renewal to take effect.
+    pub renewal_time_sec: Option<u32>,
+    pub updated_at: u32,
+}
+
+impl PeriodicBudgetInner {
+    pub(crate) fn from_budget_request(req: PeriodicBudgetRequest, updated_at: u32) -> Self {
+        Self {
+            used_budget_sat: 0,
+            max_budget_sat: req.max_budget_sat,
+            renewal_time_sec: req.renewal_time_mins.map(utils::mins_to_seconds),
+            updated_at,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct PeriodicBudget {
+    /// The amount of budget used (in satoshi) for the period
+    /// Resets once every period ([PeriodicBudget::renews_at])
+    pub used_budget_sat: u64,
+    /// The maximum budget amount allowed (in satoshi) for the period
+    pub max_budget_sat: u64,
+    /// The next timestamp at which the budget will be renewed (reset to 0)
+    pub renews_at: Option<u32>,
+    /// The latest budget update time (last reset time)
+    pub updated_at: u32,
+}
+
+impl PeriodicBudget {
+    fn from_budget_inner(b: PeriodicBudgetInner, created_at: u32) -> Self {
+        Self {
+            used_budget_sat: b.used_budget_sat,
+            max_budget_sat: b.max_budget_sat,
+            renews_at: b.renewal_time_sec.map(|t| created_at + t),
```

Comment:
should we consider last_reset_time (when exists) rather than created_at when calculating renews_at?